deepchecks/tabular/checks/model_evaluation/performance_disparity_report.py

```diff
@@ -0,0 +1,421 @@
+from deepchecks.tabular import Context, SingleDatasetCheck
+from deepchecks.core import CheckResult, ConditionCategory, ConditionResult
+from deepchecks.core.errors import DeepchecksProcessError, DeepchecksValueError
+from deepchecks.core.check_result import CheckResult
+from deepchecks.core.checks import DatasetKind
+from deepchecks.utils.typing import Hashable
+from deepchecks.utils.performance.partition import partition_column
+
+import plotly.graph_objects as go
+from plotly.subplots import make_subplots
+
+import pandas as pd
+import numpy as np
+import itertools
+
+
+class PerformanceDisparityReport(SingleDatasetCheck):
+    """
+    Check for performance disparities across subgroups, while optionally accounting for a control variable.
+
+    The check identifies "performance disparities": large performance difference for a subgroup compared to the baseline performance for the full population.
+
+    Subgroups are defined by the levels of a "protected" feature. Numerical features are first binned into quantiles, while categorical features are preserved as-is. The baseline score is the overall score when all subgroups are combined. You can add conditions to flag performance differences outside of given bounds. A visual display is also provided to identify subgroups with the largest performance disparities.
+
+    Additionally, the analysis may be separated across the levels of a "control" feature. Numerical features are binned and categorical features are re-binned into `max_number` categories. To account for the control feature, baseline scores and subgroup scores are be computed within each of its levels.
+
+    Parameters
+    ----------
+    protected_feature : Hashable
+        Feature evaluated for performance disparities. Numerical features are binned into `max_segments` quantiles. Categorical features are not transformed.
+    control_feature : Hashable, default: None
+        Feature used to group data prior to evaluating performance disparities (disparities are only assessed within the groups defined by this feature). Numerical features are binned into `max_segments` quantiles. Categorical features are re-grouped into at most `max_segments` groups if necessary.
+    scorer : str, default: None
```

Comment:
I updated it so that it has the following signature:

```
scorer : str or Tuple[str, Union[str, Callable]], default: None
```

Code to get the scorer is:
```python
if isinstance(self.scorer, str):
    scorer = context.get_single_scorer({self.scorer: self.scorer})
else:
    scorer = context.get_single_scorer(dict([self.scorer]))
```

It is convenient to be able to pass a single string as well. What do you think?

Reply:
Good catch, we should sort it later. For now I think what you did makes sense (most people will want to just pass a string)